ledger/recovering-indexer-integration-tests/src/test/suite/scala/com/digitalasset/platform/indexer/RecoveringIndexerIntegrationSpec.scala

```diff
@@ -256,13 +254,22 @@ object RecoveringIndexerIntegrationSpec {
         loggingContext: LoggingContext,
     ): ResourceOwner[ParticipantState] = {
       val metrics = new Metrics(new MetricRegistry)
-      new memory.InMemoryLedgerReaderWriter.SingleParticipantBatchingOwner(
-        ledgerId,
-        BatchingLedgerWriterConfig.reasonableDefault,
-        participantId,
-        metrics = metrics,
-        engine = Engine.DevEngine(),
-      ).map(readerWriter => new KeyValueParticipantState(readerWriter, readerWriter, metrics))
+      for {
+        dispatcher <- memory.dispatcherOwner
+        committerExecutionContext <- ResourceOwner
+          .forExecutorService(() => Executors.newCachedThreadPool())
```

Comment:
Are we sure we don't want to limit the threads?

Reply:
We could, but this is a test case, so it will be thrown away in about 5 seconds, so who cares?

Reply:
Oh sorry, you're right. 👍🏼